sql/core/src/main/scala/org/apache/spark/sql/execution/streaming/state/StateStore.scala

```diff
@@ -131,6 +141,15 @@ trait StateStore extends ReadStateStore {
   def remove(key: UnsafeRow,
     colFamilyName: String = StateStore.DEFAULT_COL_FAMILY_NAME): Unit
 
+  /**
+   * Merges the provided value with existing values of a non-null key.
```

Comment:
Maybe we want to explicitly document the contract on the behavior when performing merge against non-existing key, since all implementations should provide the same.

Reply:
Makes sense, done.